test/datatable.jl

```diff
@@ -342,6 +342,27 @@ module TestDataTable
     @test find(c -> isa(c, NullableCategoricalArray), categorical!(DataTable(A=1:3, B=4:6), [1]).columns) == [1]
     @test find(c -> isa(c, NullableCategoricalArray), categorical!(DataTable(A=1:3, B=4:6), 1).columns) == [1]
 
+    @testset "unstack nullable promotion" begin
+        dt = DataTable(Any[repeat(1:2, inner=4), repeat('a':'d', outer=2), collect(1:8)],
+                       [:id, :variable, :value])
+        udt = unstack(dt)
+        @test udt == unstack(dt, :variable, :value) == unstack(dt, :id, :variable, :value)
+        @test udt == DataTable(Any[Nullable[1, 2], Nullable[1, 5], Nullable[2, 6],
+                                   Nullable[3, 7], Nullable[4, 8]], [:id, :a, :b, :c, :d])
+        @test all(typeof.(udt.columns) .== NullableVector{Int})
+        dt = DataTable(Any[categorical(repeat(1:2, inner=4)),
+                           categorical(repeat('a':'d', outer=2)), categorical(1:8)],
+                       [:id, :variable, :value])
+        udt = unstack(dt)
+        @test udt == unstack(dt, :variable, :value) == unstack(dt, :id, :variable, :value)
+        @test udt == DataTable(Any[NullableCategoricalArray([1, 2]),
```

Comment:
`Nullable[1, 2]` and so on is enough here, `NullableCategoricalArray` is only needed in the type check.